Server/Plugins/APIDump/APIDesc.lua

```diff
@@ -10712,6 +10745,17 @@ a_Player:OpenWindow(Window);
 					},
 					Notes = "Places a block while impersonating the player. The {{OnPlayerPlacingBlock|HOOK_PLAYER_PLACING_BLOCK}} hook is called before the placement, and if it succeeds, the block is placed and the {{OnPlayerPlacedBlock|HOOK_PLAYER_PLACED_BLOCK}} hook is called. Returns true iff the block is successfully placed. Assumes that the block is in a currently loaded chunk.",
 				},
+				ReplaceOneEquippedItemTossRest =
+				{
+					Params =
+					{
+						{
+							Name = "Item",
+							Type = "cItem",
+						},
+					},
+					Notes = "Removes one item from the the current equipped item stack, and attempts to add the specified item stack back to the same slot. If it is not possible to place the item in the same slot, tries to place the specified item elsewhere in the inventory. If this is not possible, then any remaining items are tossed. Assumes that the currently equipped slot contains at least one item, behaviour is not defined in the case that it does not.",
```

Comment:
I think the behavior is quite defined - Doesn't remove anything if nothing is there; still tries to stuff Item in there.

Reply:
(same with `cInventory:ReplaceOneEquippedItem`)

Reply:
I guess I was drawing a difference between the exposed interface (behaviour is not defined) and the implementation, where we have to decide one way or the other. Do you think there is an advantage in specifying the behaviour one way or the other in the public interface?

Reply:
I'd say in this case it makes perfect sense to specify the behavior. Honestly, I think Undefined Behavior is a nightmare concept, if we can avoid it, we should.